compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -264,32 +177,139 @@ FailureOr<PadPackTiling> PadPackTiling::create(linalg::LinalgOp linalgOp) {
   // Tiling in the K-dimension is done differently TODO(newling)
   // document/reconsider.
 
-  auto maxL1Size = 16 * scaleFactor;
-  uint32_t M1 = findLargestFactor(M, maxL1Size, mPack);
-  uint32_t N1 = findLargestFactor(N, maxL1Size, nPack);
-
-  auto maxL0Size = 64 * scaleFactor;
-  uint32_t M0 = findLargestFactor(M, maxL0Size, M1);
-  uint32_t N0 = findLargestFactor(N, maxL0Size, N1);
+  if (isPackPeel) {
+    // Assume working on a 2x2 AIE array, so the ideal level 1 tile sizes should
```

Comment:
Please add a comment in the code about why we assume we're currently assuming a 2x2 array or an issue link

Reply:
Currently, for pack-peel pipeline AIR only supports 2x2 array. @erwei-xilinx will extend this to 4x4 array next.